compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -822,7 +822,10 @@ impl Visitor<'tcx> for Validator<'mir, 'tcx> {
                     let obligation = Obligation::new(
                         ObligationCause::dummy(),
                         param_env,
-                        Binder::dummy(TraitPredicate { trait_ref }),
+                        Binder::dummy(TraitPredicate {
+                            trait_ref,
+                            constness: hir::Constness::Const,
```

Comment:
This should be `NotConst` if `ccx.const_context.is_none()`

Reply:
It is not possible for the const_context to be None at this point:
```rs
fn mir_const_qualif(tcx: TyCtxt<'_>, def: ty::WithOptConstParam<LocalDefId>) -> ConstQualifs {
    let const_kind = tcx.hir().body_const_context(def.did);

    // No need to const-check a non-const `fn`.
    if const_kind.is_none() {
        return Default::default();
    }
```

Reply:
oh heh. I was looking for that but couldn't find it.